src/components/Filters/R0.js

```diff
@@ -0,0 +1,33 @@
+import React, { Component } from 'react';
+import { Slider } from 'antd';
+
+class R0 extends Component {
+
+    handleChange = (e) => {
+        // user cannot select one value as r0
+        // if (e[0] === e[1]) {return;}
+        this.props.onR0Change(e);
+    }
+
+    render() {
+        const marks = {0: '0', 1: '', 2: '', 3: '', 4: '4'};
```

Comment:
You should pull this out of the render() function and make it a constant somewhere (like between line 2 and 4). render gets called often and everytime it gets called, we are creating a new object for marks.